airflow/models/xcom_arg.py

```diff
@@ -0,0 +1,155 @@
+      op << xcomarg   (by BaseOperator code)
+
+    **Example**:
+        The moment you got result from any op (functional or regular one) you can ::
+            xcomarg = ...
+            my_op = MyOperator()
+            my_op >> xcomarg
+
+    This object can be used in legacy Operators via Jinja:
+    **Example**:
+        You can make this result to be part of any generated string ::
+            xcomarg = ...
+            op1 = MyOperator(my_text_message=f"{xcomarg}")
+            op2 = MyOperator(my_text_message=f"{xcomarg['topic']}")
+
+
+    """
+
+    def __init__(self, operator: BaseOperator, key: Optional[str] = XCOM_RETURN_KEY,
+                 runtime_sub_keys: Optional[List[str]] = None):
+        self._operator = operator
+
+        # _key is used for xcompull (key in the XCom table)
+        # _runtime_sub_keys are used for runtime access of deserialized value from XCom data provider
+        self._key = key
+        self._runtime_sub_keys = runtime_sub_keys or []
+
+    def __eq__(self, other):
+        return (self._operator == other.operator
+                and self.key == other.key
+                and self.runtime_sub_keys == other.runtime_sub_keys)
+
+    def __lshift__(self, other):
+        """
+        Implements xcomresult << op
+        """
+        self.set_upstream(other)
+        return self
+
+    def __rshift__(self, other):
+        """
+        Implements xcomresult << op
+        """
+        self.set_downstream(other)
+        return self
+
+    @property
+    def operator(self) -> BaseOperator:
+        """Operator"""
+        return self._operator
+
+    @property
+    def key(self):
+        return self._key
+
+    @property
+    def runtime_sub_keys(self):
+        return self._runtime_sub_keys
+
+    def set_upstream(self, task_or_task_list: Union[BaseOperator, List[BaseOperator]]):
+        """
+        Proxy to underlying operator set_upstream method
+        """
+        self._operator.set_upstream(task_or_task_list)
+
+    def set_downstream(
+        self, task_or_task_list: Union[BaseOperator, List[BaseOperator]]
+    ):
+        """
+        Proxy to underlying operator set_downstream method
+        """
+        self._operator.set_downstream(task_or_task_list)
+
+    def resolve(self, context: Dict) -> Any:
+        """
+        Pull XCom value for the existing arg.
+        this function likely to run during at op.execute() context
+        """
+        resolved_value = self._operator.xcom_pull(
+            context=context,
+            task_ids=[self._operator.task_id],
+            key=self.key,
+            dag_id=self._operator.dag.dag_id,
+        )
+        if not resolved_value:
+            raise AirflowException(
+                f'XComArg result from {self._operator.task_id} at {self._operator.dag.dag_id} '
+                f'with key="{self.key}"" is not found!')
+        resolved_value = resolved_value[0]
+
+        # now we can follow on user defined dereference path
```

Comment:
How did user define it? Can we add some mor information here?

Reply:
Code was removed, resolving